scripts/uosc_shared/lib/text.lua

```diff
@@ -320,37 +332,40 @@ local function opts_scale_factor(opts)
 	return (opts.italic and 1.01 or 1) * opts.size
 end
 
----@type {[boolean]: {[boolean]: {[string|number]: {[1]: number, [2]: integer}}}} | {[boolean]: {[string|number]: {[1]: number, [2]: integer}}}
-local width_cache = {}
-
----Calculate width of text with the given opts
----@param text string|number
----@return number
----@param opts {size: number; bold?: boolean; italic?: boolean}
-function text_width(text, opts)
-	if not text or text == '' then return 0 end
-
-	---@type boolean, boolean
-	local bold, italic = opts.bold or options.font_bold, opts.italic or false
-
-	if options.text_width_estimation then
-		---@type {[string|number]: {[1]: number, [2]: integer}}
-		local text_width = get_cache_stage(width_cache, bold)
-		local width_px = text_width[text]
-		if width_px and no_remeasure_required(width_px[2]) then return width_px[1] * opts_scale_factor(opts) end
-
-		local width, px = character_based_width(text, bold)
-		width_cache[bold][text] = {width, px}
-		return width * opts_scale_factor(opts)
-	else
-		---@type {[string|number]: {[1]: number, [2]: integer}}
-		local text_width = get_cache_stage(get_cache_stage(width_cache, bold), italic)
-		local width_px = text_width[text]
-		if width_px and no_remeasure_required(width_px[2]) then return width_px[1] * opts.size end
-
-		local width, px = whole_text_width(text, bold, italic)
-		width_cache[bold][italic][text] = {width, px}
-		return width * opts.size
+text_width = nil
+do
+	---@type {[boolean]: {[boolean]: {[string|number]: {[1]: number, [2]: integer}}}} | {[boolean]: {[string|number]: {[1]: number, [2]: integer}}}
+	local width_cache = {}
+
+	---Calculate width of text with the given opts
+	---@param text string|number
+	---@return number
+	---@param opts {size: number; bold?: boolean; italic?: boolean}
+	text_width = function(text, opts)
```

Comment:
You can just do

```lua
function text_width(text, opts)
```

here, it'll also make a global function, and you can delete the `text_width = nil` above.

Reply:
done